go_node_engine/cmd/root.go

```diff
@@ -49,85 +51,138 @@ func Execute() error {
 func init() {
 	rootCmd.Flags().StringVarP(&clusterAddress, "clusterAddr", "a", "localhost", "Address of the cluster orchestrator without port")
 	rootCmd.Flags().IntVarP(&clusterPort, "clusterPort", "p", 10100, "Port of the cluster orchestrator")
-	rootCmd.Flags().IntVarP(&overlayNetwork, "netmanagerPort", "n", 6000, "Port of the NetManager component, if any. This enables the overlay network across nodes. Use -1 to disable Overlay Network Mode.")
+	rootCmd.Flags().IntVarP(&netmanagerPort, "netmanagerPort", "n", 0, "Port of the NetManager component (deprecated).")
+	rootCmd.Flags().StringVarP(&overlayNetwork, "overlayNetwork", "o", "default", "Options: default,disabled,custom:<path>. <path> points to the overlay component socket.")
 	rootCmd.Flags().BoolVarP(&unikernelSupport, "unikernel", "u", false, "Enable Unikernel support. [qemu/kvm required]")
-	rootCmd.Flags().StringVarP(&logDirectory, "logs", "l", "/tmp", "Directory for application's logs")
-	// Addons
-	rootCmd.Flags().BoolVar(&imageBuilder, "image-builder", false, "Checks if the host has QEMU (apt's qemu-user-static) installed for building multi-platform images.")
-	rootCmd.Flags().BoolVar(&flopsLearnerSupport, "flops-learner", false, "Enables the ML-data-server sidecar for data collection for FLOps learners.")
+	rootCmd.Flags().StringVarP(&logDirectory, "logs", "l", config.DEFAULT_LOG_DIR, "Directory for application's logs")
+	rootCmd.Flags().BoolVarP(&detatched, "detatch", "d", false, "Run the NodeEngine in the background (daemon mode)")
+	rootCmd.Flags().StringVarP(&certFile, "certFile", "c", "", "Path to certificate for TLS support")
+	rootCmd.Flags().StringVarP(&keyFile, "keyFile", "k", "", "Path to key for TLS support")
 }
 
-func startNodeEngine() error {
-	// set log directory
-	model.GetNodeInfo().SetLogDirectory(logDirectory)
+func nodeEngineDaemonManager() error {
 
-	// connect to container runtime
-	runtime := virtualization.GetContainerdClient()
-	defer runtime.StopContainerdClient()
-
-	if unikernelSupport {
-		unikernelRuntime := virtualization.GetUnikernelRuntime()
-		defer unikernelRuntime.StopUnikernelRuntime()
+	if _, err := os.Stat(CONF_FILE); err != nil {
+		// read cluster configuration if not present or new value set
+		err := defaultConfig()
+		if err != nil {
+			return err
+		}
 	}
 
-	if imageBuilder {
-		cmd := exec.Command("dpkg", "-s", "qemu-user-static")
-		output, err := cmd.Output()
-		if err != nil || !strings.Contains(string(output), "ok installed") {
-			logger.ErrorLogger().Fatalf("Unable to find qemu-user-static apt package for multi-platform image-builder: %v\n", err)
+	if clusterAddress != "localhost" {
+		// read cluster configuration if not present or new value set
```

Comment:
same question as above and look below

Reply:
```suggestion
		// read cluster configuration if not present or no other value provided
```

Reply:
Seems fine - I just found the first wording a bit strange - choose what fits the scenario best